lib/src/configuration.cc

```diff
@@ -30,33 +30,34 @@ namespace lth_log = leatherman::logging;
 
 #ifdef _WIN32
     namespace lth_w = leatherman::windows;
-    static const fs::path DATA_DIR = []() {
+    static fs::path getWinPath(int folder_csidl) {
         wchar_t szPath[MAX_PATH+1];
-        if (FAILED(SHGetFolderPathW(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+        if (FAILED(SHGetFolderPathW(NULL, folder_csidl, NULL, 0, szPath))) {
             throw Configuration::Error {
                 (boost::format("failure getting Windows AppData directory: %1%")
                     % lth_w::system_error()).str() };
         }
-        return fs::path(szPath) / "PuppetLabs" / "pxp-agent";
-    }();
+        return fs::path { szPath };
+    }
+
+    static const fs::path DATA_DIR = getWinPath(CSIDL_COMMON_APPDATA)
+                                        / "PuppetLabs" / "pxp-agent";
+    static const fs::path BIN_DIR = getWinPath(CSIDL_PROGRAM_FILES)
+                                        / "Puppet Labs" / "Puppet" / "pxp-agent";
```

Comment:
I think @Iristyle and I agree that it'd be better to use a relative path to the binary, as in http://stackoverflow.com/questions/2647429/c-windows-path-to-the-folder-where-the-executable-is-located.